BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/physicsOfResultsPUEO.py

```diff
@@ -55,85 +57,147 @@
                 print(os.path.join(path, name))
                 
                 try:
                     # Open the root file and assign the trees to variables
                     fileName = os.path.join(path, name)
                     IceFinalFile = ROOT.TFile(fileName)
                     
-                    nuWeights = []
-                    nuPasses = []
 
                     allTree = IceFinalFile.allTree
                     allEvents = allTree.GetEntries()
-
+                    print(allEvents)
                     TotalEvents[energy].append(allEvents)
+                    
+                except:
+                    print("Error opening file")
+                    continue
+                
+            elif fnmatch(name, pass_tree_pattern):
+                print(os.path.join(path, name))
+                
+                try:    
+                    
+                    fileName = os.path.join(path, name)
+                    IceFinalFile = ROOT.TFile(fileName)
+                    
                     passTree = IceFinalFile.passTree
                     passEvents = passTree.GetEntries()
+                    
+                    nuWeights = []
+                    nuPasses = []
+                    print(passEvents)
+                    
                 except:
                     print("Error opening file")
                     continue
-                success_runs.append(fileName)
+                
+                
+
+                passTree.GetEvent(0)
+                #Create a canvas for the TGraphs
+                c = ROOT.TCanvas("c", "c", 1000, 1000)
+                c.Divide(1, 2)
+                c.cd(1)
+                signalAtDetector = passTree.event.signalAtDetector
+                # Add titles and axes to the TGraph and then save as a png
+                signalAtDetector.SetTitle("Signal at Detector for {} EeV Neutrinos".format(g.energy))
+                signalAtDetector.GetXaxis().SetTitle("Time (ns)")
+                signalAtDetector.GetYaxis().SetTitle("Voltage (V)")
+                signalAtDetector.SetLineColor(1)
+                signalAtDetector.SetLineWidth(2)
+                signalAtDetector.SetMarkerStyle(20)
+                signalAtDetector.SetMarkerSize(0.5)
+                signalAtDetector.SetMarkerColor(1)
+                signalAtDetector.Draw()
+                c.cd(2)
+                signalAtSource = passTree.event.signalAt1m
+                signalAtSource.SetTitle("Signal at 1m for {} EeV Neutrinos".format(g.energy))
+                signalAtSource.GetXaxis().SetTitle("Time (ns)")
+                signalAtSource.GetYaxis().SetTitle("Voltage (V)")
+                signalAtSource.SetLineColor(1)
+                signalAtSource.SetLineWidth(2)
+                signalAtSource.SetMarkerStyle(20)
+                signalAtSource.SetMarkerSize(0.5)
+                signalAtSource.SetMarkerColor(1)
+                signalAtSource.Draw()
```

Comment:
Instead of duplicating this part, you could do:
```py
            for plot in [signalAtDetector, signalAtSource]:
                plot.GetXaxis().SetTitle("Time (ns)")
                plot.GetYaxis().SetTitle("Voltage (V)")
                plot.SetLineColor(1)
                plot.SetLineWidth(2)
                plot.SetMarkerStyle(20)
                plot.SetMarkerSize(0.5)
                plot.SetMarkerColor(1)
                plot.Draw()
```

Reply:
I decided to just delete the signalAtSource, as it isn't interesting to us and could waste some computing time.